components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1831,40 +2009,38 @@ String[] filterScopeValidators(OAuthConsumerAppDTO application) throws IdentityO
     /**
      * Get the IdToken Encryption Method registered by the user and filter the allowed one.
      *
-     * @param application Application user have registered
+     * @param encryptionMethod Encryption method sent in the registration request.
      * @return idTokenEncryptionMethod
      * @throws IdentityOAuthAdminException Identity OAuthAdmin exception.
      */
-    private String filterIdTokenEncryptionMethod(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
+    private String filterEncryptionMethod(String encryptionMethod) throws IdentityOAuthAdminException {
```

Comment:
This method and `filterEncryptionAlgorithms` method refactoring is confusing. Even though the method is refactored to be made generic in both methods id token related server configs are only read (getSupportedIdTokenEncryptionMethods and getSupportedIdTokenEncryptionAlgorithm)

Reply:
changed to give a the config name as a parameter